com.siemens.ros-sharp/Runtime/Libraries/RosBridgeClient/MessageTypes/ROS2/Geometry/msg/Quaternion.cs

```diff
@@ -16,10 +16,10 @@ public class Quaternion : Message
         public const string RosMessageName = "geometry_msgs/msg/Quaternion";
 
         //  This represents an orientation in free space in quaternion form.
-        public double x = 0;
-        public double y = 0;
-        public double z = 0;
-        public double w = 1;
+        public double x { get; set; }
```

Comment:
Declaring an auto-property will fix the serialization problem due to the hidden backing field, but this is an auto-generated message so it should not be modified; rather, the `Message Generator` should be modified. Also, `geometry_msgs/msg/Quaternion` includes default values by message definition. 
```suggestion
        public double x { get; set; } = 0;
```